### ISSUE TITLE
Bug: @typescript-eslint/consistent-type-imports The rule does not correctly handle `@jsx*` directives.

### ISSUE THREAD

**Issue by iclanton** [typescript-eslint/typescript-eslint#11749]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.3.3&fileType=.tsx&code=PQKhAIAECsGcA9zQPYEsB24TAFCgjAgGIBOAhgOZJqbY46oC2ADsiQC7gDe1G4AvuABmJZI3ABydMgAmAUwBczMuwAWEgNz058Vh2EBXdAGN2qZJgAKJOWVMBhMa3Rz07ABQBKBeABKc9gMSdAAVAE9mOQAedgi5ZCFedAA%2BbhxwcBtA4PAo5KjgZK1%2BHCA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWpQC2xDtHyoMkRNGhTI4MAF8QaoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkAViQB4VSboCG0ekANODZBJ5MyLi2EBXdHxpRkJWKnZ50AEwCysVRIxlKk6f1rwtO9C0joSRdI278AviAdA&tokens=false

### Repro Code

```TypeScript
/** @jsx join */
/** @jsxFrag join */

import { join } from 'node:path';

export function PreactComponent(): ReturnType<typeof join> {
  return <></>;
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/consistent-type-imports": "error"
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "jsx": "react"
  }
}
```

### Expected Result

I expect there to be no issues with the provided example.

### Actual Result

>All imports in the declaration are only used as types. Use `import type`. 4:1 - 4:34

### Additional Info

In this example, the JSX elements are handled by React by default, but specific modules use something else. In the practical repro where we're seeing this, it's `preact`, but in this example, I'm using something available from the standard `node` types.

Link to the `@jsx` directive documentation: https://www.typescriptlang.org/tsconfig/#jsxFactory

**Comment by bradzacher** [comment]
Our scope analysis infra does not scan for jsx pragmas in the code. JSX pragmas see very, very, very low usage and instead people leverage the tsconfig compiler options. It's rare that a codebase needs a handful of files to use a different set of JSX configs and instead it's a codebase-wide change.

You can use the parser options to mirror the tsconfig settings so that the scope analysis respects the jsx config you're using:

https://typescript-eslint.io/packages/parser#jsxfragmentname
https://typescript-eslint.io/packages/parser#jsxpragma

**Comment by JoshuaKGoldberg** [comment]
Note that there is also pending work to refactor globals for ESLint 10: #11762.

I also am tentatively not in favor of accepting this issue. It's a bummer for folks using `@jsx*` 😕 but, as Brad said, it really is an exceptionally rare case. @iclanton can you post more info about the full case you alluded to? Maybe it'll be very motivating?

**Comment by JoshuaKGoldberg** [comment]
Closing out as not receiving additional requested info. If anybody can describe the full case alluded to here, please do post back - in this issue if it's unlocked, or a new one otherwise. Cheers!
